File: synapse-transport/src/main/java/com/synapse/transport/feature/experiment/dto/response/ExperimentHomeResponse.java
package com.synapse.transport.feature.experiment.dto.response;

import java.time.LocalDateTime;
import java.util.UUID;

import com.synapse.transport.common.enums.ExperimentStatus;
import com.synapse.transport.common.enums.ExperimentTaskType;
import com.synapse.transport.common.enums.InferenceStatus;
import com.synapse.transport.common.enums.ModelStatus;
import com.synapse.transport.common.enums.ConfigStatus;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ExperimentHomeResponse {
    private UUID workspaceId;
    private UUID experimentId;

    private String name;
    private String description;

    private ExperimentTaskType taskType;
    private ExperimentStatus status;

    private ConfigStatus datasetReady;
    private ModelStatus modelReady;
    private InferenceStatus inferenceReady;

    private String createdBy;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: synapse-ui/src/types/api/experiment/experiment-home.response.ts
import type {
  ConfigStatus,
  ExperimentStatus,
  ExperimentTaskType,
  InferenceStatus,
  ModelStatus,
} from "../../enums/transport.enums";

export interface ExperimentHomeResponse {
  workspaceId: string;
  experimentId: string;
  name: string;
  description: string;
  taskType: ExperimentTaskType;
  status: ExperimentStatus;
  datasetReady: ConfigStatus;
  modelReady: ModelStatus;
  inferenceReady: InferenceStatus;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
}
